crates/polars-ops/src/series/ops/is_in.rs

```diff
@@ -362,20 +362,36 @@ fn is_in_cat(ca_in: &CategoricalChunked, other: &Series) -> PolarsResult<Boolean
         },
         DataType::String => {
             let ca_other = other.str().unwrap();
-            let categories = ca_in.get_rev_map().get_categories();
-
+            let rev_map = ca_in.get_rev_map();
+            let categories = rev_map.get_categories();
             let others: PlHashSet<&str> = ca_other.downcast_iter().flatten().flatten().collect();
+            let mut set = PlHashSet::with_capacity(std::cmp::min(categories.len(), ca_other.len()));
```

Comment:
Instead of using a HashSet, would it be faster to allocate a vector v of values 0...max(idx)-1 (with True at the loc of each global idx), and then instead of `set.contains` at the end, simply return `v[s]` for each `s` in `ca_in.physical()`? In other words, suppose `ca_in = ["c", "c", "f", "h", "f", "c"]`:

```
gobal map:     | a | b | c | d | e | f | g | h |
               |---|---|---|---|---|---|---|---|
               | 0 | 1 | 2 | 3 | 4 | 5 | 6 | 7 |   

local contain: | 0 | 0 | 1 | 0 | 0 | 1 | 0 | 1 |   // x in global for x in ca_in.unique()
```

Then for each value of `ca_in` we could just do a lookup of `local_contain[s]` for each `s` in `ca_in`.